corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -578,11 +578,15 @@ def get_wrapped_ledger_values(domain, case_ids, section_id, entry_ids=None, pagi
     return [StockLedgerValueWrapper.wrap(row) for row in query.run().hits]
 
 
-def get_total_records(domain, case_ids, section_id, entry_ids=None):
+def get_products_count(domain, case_ids, section_id, entry_ids=None):
+    return len(get_product_ids_from_ledgers(domain, case_ids, section_id, entry_ids))
+
+
+def get_product_ids_from_ledgers(domain, case_ids, section_id, entry_ids=None):
     query = LedgerES().domain(domain).section(section_id).case(case_ids)
     if entry_ids:
         query = query.entry(entry_ids)
-    return query.count()
+    return set(query.values_list('entry_id', flat=True))
```

Comment:
Does this report only show products that have entries associated with them?  I'd expect it to include all products and just show `0` or `---` for the ones without data.  I understand changing that may be out of scope if you're not confident.

Reply:
I kept things according to the original behaviour even though the report earlier was broken.
Plus including blanks for missing products would need to relook at the [logic](https://github.com/dimagi/commcare-hq/pull/21101/files#diff-007cbd344b264b88f44398660b42183aR216) where data is being built and iterate differently.
So keeping this.